### ISSUE TITLE
Extract Body

### ISSUE THREAD

**Issue by CameliaSNCF** [MAIF/melusine#123]
**Python version** : 3.8.10

**Melusine version** : 2.3.2

**Operating System** : Windows 

Hello, I have an issue regarding the extraction of the segmenting of the body to clean, during the segmentation Melusine tags the CC of the email as body resulting in the body_header_extract funtion to consider the CC as the body to clean.

![segmentingBody](https://user-images.githubusercontent.com/95216688/148962607-3e6e41d8-e736-471a-aeca-c729b5be93c0.png)

![extractBody](https://user-images.githubusercontent.com/95216688/148962673-0762c031-a89f-4a58-8fe3-6fcca6aaa925.png)

In the exemple above, you see the part that was selected as the body (yellow) does not correspond to the actual body of the email (between blue)

Thank you for your help.
Best regards.
Camelia

**Comment by TFA-MAIF** [comment]
Could you share the different steps of your pipeline? By that I mean all the steps from your original email to structured_body.

Best regards

**Comment by CameliaSNCF** [comment]
Hello, our data doesn't come in the same format as yours so here is how it starts  :

![CorpsDeTexte1](https://user-images.githubusercontent.com/95216688/149745094-596f8725-61f2-4b3d-8b4c-ab39bca471e6.png)
![CorpsDeTexte2](https://user-images.githubusercontent.com/95216688/149745109-f66eeaa0-c6c7-4014-bd41-8a07ca6945f3.png)

I then try to eliminate the parts that are useless and try to get the email as close to your format as I can : 

![body3](https://user-images.githubusercontent.com/95216688/149745170-f6a1c191-c726-4cf9-80d1-c0d876fac0f8.png)

then I apply the segmenting pipeline : 

![Segmenting3](https://user-images.githubusercontent.com/95216688/149745250-3aeec01b-9ecb-435b-98ef-e2a268529d4f.png)

And finally the cleaning pipeline :

![cleaning](https://user-images.githubusercontent.com/95216688/149745303-f01d85f1-46eb-41fe-847c-00db82520cff.png)

**Comment by CameliaSNCF** [comment]
Is this enough information to understand the problem ?

**Comment by TFA-MAIF** [comment]
Hello, 

Yes thank you.
As your email has a specific format you need to adapt regexs that parse the historic. Because the default regex to split messages stopped after the ";" of the "A: " field. You want it to go further.
You will need to adapt regex from the key ["build_historic"]["transition_list"] in conf.json to match your format.

Best regards.

**Comment by CameliaSNCF** [comment]
Hello, thank you very much for your answer, it was very useful.
I changed the regex where you indicated it, as the ';' is only used as a separator for the recipients of the email and I am interested in the line break, I took the ';' off the regex and it works.

![regex](https://user-images.githubusercontent.com/95216688/150108882-f103d062-3c15-4a11-9375-fd2826f8856f.png)

Thank you again.
Best regards

Camelia
